state/protocol/badger/snapshot.go

```diff
@@ -47,80 +45,16 @@ func (s *Snapshot) Head() (*flow.Header, error) {
 }
 
 // QuorumCertificate (QC) returns a valid quorum certificate pointing to the
-// header at this snapshot. With the exception of the root block, a valid child
-// block must be which contains the desired QC. The sentinel error
-// state.NoChildBlockError is returned if the the QC is unknown.
-//
-// For root block snapshots, returns the root quorum certificate. For all other
-// blocks, generates a quorum certificate from a valid child, if one exists.
+// header at this snapshot.
+// The sentinel error storage.ErrNotFound is returned if the QC is unknown.
 func (s *Snapshot) QuorumCertificate() (*flow.QuorumCertificate, error) {
-
-	// CASE 1: for the root block, return the root QC
-	root, err := s.state.Params().Root()
-	if err != nil {
-		return nil, fmt.Errorf("could not get root: %w", err)
-	}
-
-	if s.blockID == root.ID() {
-		var rootQC flow.QuorumCertificate
-		err := s.state.db.View(operation.RetrieveRootQuorumCertificate(&rootQC))
-		if err != nil {
-			return nil, fmt.Errorf("could not retrieve root qc: %w", err)
-		}
-		return &rootQC, nil
-	}
-
-	// CASE 2: for any other block, generate the root QC from a valid child
-	child, err := s.validChild()
+	qc, err := s.state.qcs.ByBlockID(s.blockID)
 	if err != nil {
-		return nil, fmt.Errorf("could not get valid child of block %x: %w", s.blockID, err)
+		return nil, fmt.Errorf("could not retrieve quorum certificate for (%x): %w", s.blockID, err)
 	}
-
-	// sanity check: ensure the child has the snapshot block as parent
-	if child.ParentID != s.blockID {
-		return nil, fmt.Errorf("child parent id (%x) does not match snapshot id (%x)", child.ParentID, s.blockID)
-	}
-
-	// retrieve the full header as we need the view for the quorum certificate
-	head, err := s.Head()
-	if err != nil {
-		return nil, fmt.Errorf("could not get head: %w", err)
-	}
-
-	qc := &flow.QuorumCertificate{
-		View:          head.View,
-		BlockID:       s.blockID,
-		SignerIndices: child.ParentVoterIndices,
-		SigData:       child.ParentVoterSigData,
-	}
-
 	return qc, nil
 }
 
-// validChild returns a child of the snapshot head. Any valid child may be returned.
-// Subsequent calls are not guaranteed to return the same child.
-// Since blocks are fully validated before insertion to the state, all stored child
-// blocks are valid and may be returned.
-//
-// Error returns:
-//   - state.NoChildBlockError if no valid child exists.
```

Comment:
`NoChildBlockError` is no longer used - we can remove it

https://github.com/onflow/flow-go/blob/5c4e877dc3bdca7d8617bfee7f25c7c95e9660ae/state/errors.go#L93-L116